### DISCUSSION TITLE
Does trash autodelete work?

### DISCUSSION THREAD

**Discussion by ViRb3** [dani-garcia/vaultwarden#1705] [Q&A] [answered]
Bitwarden has now enabled autodelete of trash items older than 30 days: https://bitwarden.com/help/article/releasenotes/#2021-05-11

I wanted to check how vaultwarden deals with this, but I found somewhat conflicting information:

In the template file, a cron job is present and enabled by default:
https://github.com/dani-garcia/vaultwarden/blob/47e8aa29e1e8b58a7534b025b32483c48943b136/.env.template#L77-L79 

However, no sight of how old items have to be to get deleted. In the main config file:

https://github.com/dani-garcia/vaultwarden/blob/47e8aa29e1e8b58a7534b025b32483c48943b136/src/config.rs#L357-L360

It seems like the autodelete will be disabled if no days are defined. From my understanding, by default indeed no days are defined. Does this mean that the trash will never be autodeleted? If that's the case, why is a cron job running?

Thanks

**Comment by jjlin** [comment, marked as answer]
You can set the `TRASH_AUTO_DELETE_DAYS` env var to enable auto-deletion. In general, any config item in config.rs can be set via an env var of the same name (except all caps).

`TRASH_AUTO_DELETE_DAYS` was inadvertently missing, but will be added in #1706.

**Reply by ViRb3** [reply]
Thank you!
